### PR TITLE
Add test scripts

### PR DESCRIPTION
**Pull request by amishas157**
Added test script which checks following test cases:
- An error is thrown on giving input file as null .
- The output file name is as expected when it is not passed as argument
- Output is as expected on a test geojson file.

### CODE REVIEW

**Comment by arunasank** [comment]
@amishas157 This is very close to shipping! 🎉 

Couple of things:
- Might want to rewrite `geojsonstreammerge` in [camel case](https://en.wikipedia.org/wiki/CamelCase)
- You can remove all the comments that explain something, like in https://github.com/geohacker/geojson-stream-merge/pull/10/files#diff-c1129c8b045390789fa8ff62f2c6b4a9R3
- You might want to choose a test fixture that has more than one `FeatureCollection` (can have just one feature in each `FeatureCollection`) since the script takes in many line delimited feature collections and makes one `FeatureCollection`

So something like

```
{ "type": "FeatureCollection", "features": [<feature1>, <feature2>]}
{ "type": "FeatureCollection", "features": [<feature3>]}
```

And see if the expected output is like 

```
{ "type": "FeatureCollection", "features": [<feature1>, <feature2>, <feature3>]}
```

Also, nice job with modifying https://github.com/geohacker/geojson-stream-merge/pull/10/files#diff-168726dbe96b3ce427e7fedce31bb0bcR43 to be able to test the `outputFile` and its contents. 😅

File: test/test.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  	assert.equals(data, "test-merged.geojson", "Awesome! Test Case passes");
+  	assert.end();
+  })
+  
+});
+
+tape('check if the output is as per expected', function(assert) {
+  geojsonstreammerge("test.geojson", "output.geojson",function(err,data) {
+  	var output = fs. readFileSync('output.geojson', 'utf8');
+    var testOutput = fs. readFileSync('testOutput.geojson', 'utf8');
```
Comment (geohacker) [inline test/test.js]: @amishas157 could you indent this line to match rest of the file?

**Comment by geohacker** [comment]
@amishas157 nice work! I'd make one more addition to the test that compares the output - use https://github.com/mapbox/geojsonhint to validate the actual geojson that's returned by the function.

Thanks for doing this!

File: test/test.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  geojsonstreammerge(null, null,function(err,data) {
+  	assert.ifError(!err);
+  	assert.end();
+  })
+  
+});
+
+tape('Check output file name when output file name argument is empty', function(assert) {
+  geojsonstreammerge("test.geojson", null,function(err,data) {
+  	assert.equals(data, "test-merged.geojson", "Awesome! Test Case passes");
```
Comment (geohacker) [inline test/test.js]: @amishas157 you can just say 'ok, valid geojson' or something as the string than 'Awesome' :)

Comment (geohacker) [inline test/test.js]: or valid filename.

**Comment by geohacker** [comment]
@amishas157 - this is coming along pretty well! Couple of notes:
- the tests were failing for me because the file paths weren't absolute. This is generally a good idea. You can use the node helper [__dirname](https://nodejs.org/api/all.html#globals_dirname) + '/filename.geojson'
- can you remove [output.geojson](https://github.com/amishas157/geojson-stream-merge/blob/tests/test/test.js#L23) after the test run? this caused some trouble because for every consequent test run, geojson-stream-merge appended the same data, and tests failed. You can try a tear down method for this https://gist.github.com/substack/9561717
- for geojsonAssert, i had to wrap it in an assert because that's is what tape's assertion:

```
tape('is geojson valid?', function(assert) {
  var output = fs.readFileSync(__dirname + '/output.geojson', 'utf8');
  assert.equals(geojsonAssert(output), undefined, "ok, valid geojson");
  assert.end();
});
```

**Comment by arunasank** [comment]
@amishas157 This looks 💥 

Since we changed the callback in https://github.com/geohacker/geojson-stream-merge/pull/10/files#diff-168726dbe96b3ce427e7fedce31bb0bcR43, do you also want to update the callback function in [the API usage example](https://github.com/amishas157/geojson-stream-merge/blob/tests/README.md#api) in the README.md file with the [new callback form](https://github.com/amishas157/geojson-stream-merge/blob/tests/index.js#L43)?

Just adding a few more (very minor) notes 😅 :
- Might want to rename the input file as `test` instead of `test.geojson` since it's not a valid geojson.
- Would be nice if the input and the output geojson are named similarly so either `testInput` and `testOutput.geojson` or `input` and `output.geojson`

Let's also set you up with a linter and get this code to ✨

**Comment by geohacker** [comment]
@amishas157 nice work! I pulled your changes into a branch and the tests don't pass for me because of this line: https://github.com/geohacker/geojson-stream-merge/pull/10/files#diff-c1129c8b045390789fa8ff62f2c6b4a9R12 

Looking at the docs - https://nodejs.org/api/assert.html#assert_assert_iferror_value:

> Throws value if value is truthy. 

So `!err` will make this truthy when we get `null` and fail the test. I think you should leave this as: `assert.ifError(err)`

Let me know if I'm missing something!
Thank you for working on this.

**Comment by amishas157** [comment]
@geohacker Yeah, that's right. The test is failing due to `!null -> true`. Corrected it. Thanks.

**Comment by geohacker** [comment]
@amishas157 great, thank you! Do you have more changes coming in or can I merge this?

**Comment by amishas157** [comment]
@geohacker It's good to merge. All the tests pass now. 6/6

**Comment by geohacker** [comment]
Awesome!

🙇 @amishas157 @arunasank

**Pull request merged**